packages/web3-shared/evm/constants/descriptors.ts

```diff
@@ -312,6 +312,21 @@ export const PROVIDER_DESCRIPTORS: Array<ProviderDescriptor<ChainId, ProviderTyp
         shortenLink: 'mathwallet.org',
         downloadLink: 'https://mathwallet.org/en-us/#extension',
     },
+    {
+        ID: `${PLUGIN_ID}_opera`,
+        providerAdaptorPluginID: PLUGIN_ID,
+        type: ProviderType.Opera,
+        name: 'Opera',
+        icon: new URL('../assets/opera.png', import.meta.url),
+        enableRequirements: {
+            supportedChainIds: getEnumAsArray(ChainId).map((x) => x.value),
+            supportedEnhanceableSites: getEnumAsArray(EnhanceableSite).map((x) => x.value),
+            supportedExtensionSites: getEnumAsArray(ExtensionSite).map((x) => x.value),
```

Comment:
Not all EVM-compatible chains are supported.

![image](https://user-images.githubusercontent.com/52657989/178930578-43a816b8-8eb8-44f5-969f-c19df4e76937.png)